Source/WebCore/page/Frame.cpp

```diff
@@ -180,6 +180,11 @@ Frame::Frame(Page& page, HTMLFrameOwnerElement* ownerElement, UniqueRef<FrameLoa
 
 void Frame::init()
 {
+    // This hack enables the experimental WebAPIsInShadowRealm WebCore feature to turn on ShadowRealms in JSC,
```

Comment:
This should bot be right. Frame construction is too late to set these option values. You need to set it before creating any of JSC VM. Thus almost prologue of process launch.
See useSharedArrayByffer config for reference.

Reply:
I mean, it does work, but I'll look into that. Cheers 🙏

Reply:
I see -- so the plan is to disallow what I've done here in the future, makes sense.

Reply:
I'm having another go at this -- the approach I'm attempting is:

Mac port:
- Add an xpc initializer key called 'enable-shadow-realm', and turn it on during the process launch if the WebAPIsInShadowRealms preference is eanbled.

The question is, how can I determine that WebAPIsInShadowRealms is enabled. Some other code seems like it looks at the preferences for every page in the WebProcessProxy page map, and treat the setting as enabled if it's enabled anywhere. This doesn't seem like the worst thing, so maybe this is ok. But I thought I'd ask in case there's a preferred way to go about it, or the other uses seem to solve a different issue.

For other ports: I haven't really gotten anywhere, as there aren't really any examples of this happening cross-platform. I expect if I can make it work on Mac, I can do it anywhere, but maybe not.